controllers/core/flagsourceconfiguration/controller.go

```diff
@@ -104,6 +152,124 @@ func (r *FlagSourceConfigurationReconciler) Reconcile(ctx context.Context, req c
 	return r.finishReconcile(nil, false)
 }
 
+func (r *FlagSourceConfigurationReconciler) handleKubeProxy(ctx context.Context) error {
+	exists, err := r.doesKubeProxyExist(ctx)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return r.deployKubeProxy(ctx)
+	}
+	return nil
+}
+
+func (r *FlagSourceConfigurationReconciler) deployKubeProxy(ctx context.Context) error {
+	r.Log.Info("deploying the kube-flagd-proxy")
+	if err := r.Client.Create(ctx, r.newFlagdKubeProxyManifest()); err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+	r.Log.Info("deploying the kube-flagd-proxy service")
+	if err := r.Client.Create(ctx, r.newFlagdKubeProxyServiceManifest()); err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+func (r *FlagSourceConfigurationReconciler) newFlagdKubeProxyServiceManifest() *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      KubeProxyServiceName,
+			Namespace: r.KubeProxyConfig.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app.kubernetes.io/name": KubeProxyDeploymentName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "flagd-kube-proxy",
```

Comment:
Might be worth putting `flagd-kube-proxy` into a public const somewhere. We use it in some log lines elsewhere.